### PR TITLE
CI: default py3.8

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

- add python 3.10
- set default python 3.8

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/ci_test-full.yml [hunk 1]
```diff
@@ -17,13 +17,11 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-20.04, macOS-10.15, windows-2019]
-        python-version: [3.6, 3.8, 3.9]
+        python-version: ["3.6", "3.8", "3.10"]
```
Comment (justusschock) [inline .github/workflows/ci_test-full.yml:20]:
```suggestion
        python-version: ["3.6", "3.8", "3.9"]
```

**Review by justusschock — CHANGES_REQUESTED** [review]
Can only use python 3.9!
Torch Binaries are not yet available for 3.10 (neither on pypi nor on https://download.pytorch.org/whl/cpu/torch_stable.html)

**Comment by Borda** [comment]
> Can only use python 3.9! Torch Binaries are not yet available for 3.10

yep, I found it too, so this will hang tll they start supporting it :] otherwise, this PR does not make much sense if we still keep 3.9 as the latest python and the PR is called add 3.10 support :]

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
ok, the problem is that `pypesq` needs NumPy 1.20 but python 3.10 does not have it...

**Comment by Borda** [comment]
with running the `latest` I get:
```
tests\helpers\testers.py:380: in run_functional_metric_test
    _functional_test(
tests\helpers\testers.py:291: in _functional_test
    sk_result = sk_metric(preds[i].cpu(), target[i].cpu(), **extra_kwargs)
tests\audio\test_si_sdr.py:54: in speechmetrics_si_sdr
    metric = speechmetrics_sisdr(preds[i, j], target[i, j], rate=16000)
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\speechmetrics\__init__.py:117: in __call__
    result_metric = metric.test(*files, array_rate=rate)
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\speechmetrics\__init__.py:29: in test
    import resampy
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\resampy\__init__.py:7: in <module>
    from .core import *
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\resampy\core.py:9: in <module>
    from .interpn import resample_f
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\resampy\interpn.py:4: in <module>
    import numba
C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\numba\__init__.py:200: in <module>
    _ensure_critical_deps()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

        if numpy_version < (1, 18):
            raise ImportError("Numba needs NumPy 1.18 or greater")
        elif numpy_version > (1, 21):
>           raise ImportError("Numba needs NumPy 1.21 or less")
E           ImportError: Numba needs NumPy 1.21 or less
```

**Comment by Borda** [comment]
seem #949 would be a very needed update for this case

**Comment by Borda** [comment]
now it fails to build the `numpy` from the source... :/

File: src/torchmetrics/utilities/imports.py [hunk 2]
```diff
@@ -112,7 +112,6 @@ def _compare_version(package: str, op: Callable, version: str) -> Optional[bool]
 _SCIPY_AVAILABLE: bool = _package_available("scipy")
 _TORCH_FIDELITY_AVAILABLE: bool = _package_available("torch_fidelity")
 _LPIPS_AVAILABLE: bool = _package_available("lpips")
```
Comment (Borda) [inline src/torchmetrics/utilities/imports.py:114]:
```suggestion
_LPIPS_AVAILABLE: bool = _package_available("lpips")
_PYCOCOTOOLS_AVAILABLE: bool = _package_available("pycocotools")
```

**Pull request merged**